File: C++/2021/1stOct_Amaan-Farooq.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> SieveOfEratosthenes(vector<int> r)
{
    int n = r[r.size()-1];
    bool prime[n + 1];
    memset(prime, true, sizeof(prime));
    prime[0] = false;
    prime[1] = false;
    for (int p = 2; p * p <= n; p++)
    {
        if (prime[p] == true)
        {
            for (int i = p * p; i <= n; i += p)
                prime[i] = false;
        }
    }

    for(int i=0; i<r.size(); i++)
        if(prime[r[i]]==true || r[i]%5==0)
            r[i] = 0;
    return r;
}

vector<int> printFibonacciNumbers(int n)
{
    vector<int> r;
    int f1 = 0, f2 = 1, i;

    if (n < 1)
        return r;
    for (i = 1; i < n; i++) {
        r.push_back(f2);
        int next = f1 + f2;
        f1 = f2;
        f2 = next;
    }
    return r;
}

int main()
{
    int n;
    cout<<"Enter number of terms required:";
    cin>>n;
    vector<int> r = printFibonacciNumbers(n+1);
    r = SieveOfEratosthenes(r);
    for(int i=0; i<r.size(); i++)
        cout<<r[i]<<" ";
    return 0;
} 

File: Python/2021/1stOct_Amaan-Farooq.py
def is_prime(num):
    if num > 1:
        for i in range(2, num // 2 + 1):
            if (num % i) == 0:
                return False
        else:
            return True
    else:
        return False

n = int(input("Enter the number of terms needed"))

n1, n2 = 1, 1 
cnt = 0

if n == 1:
    print(n1)
else:
    while cnt < n:
        if not is_prime(n1) and n1 % 5 != 0:
            print(n1, end=' ')
        else:
            print(0, end=' ')
        sum = n1 + n2
        n1 = n2
        n2 = sum
        cnt += 1 
